src/layers/core/point-density-grid-layer/point-density-grid-layer.js

```diff
@@ -0,0 +1,138 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer} from '../../../lib';
+import GridLayer from '../grid-layer';
+
+import {pointToDensityGridData} from './grid-aggregator';
+
+const defaultCellSize = 1;
+const defaultColorRange = [
+  [255, 255, 178],
+  [254, 217, 118],
+  [254, 178, 76],
+  [253, 141, 60],
+  [240, 59, 32],
+  [189, 0, 38]
+];
+const defaultElevationRange = [0, 10];
+
+const defaultProps = {
+  cellSize: defaultCellSize,
+  colorRange: defaultColorRange,
+  elevationRange: defaultElevationRange
+};
+
+function noop() {}
+
+function _needsReProjectPoints(oldProps, props) {
+  return oldProps.cellSize !== props.cellSize;
+}
+
+function _ordinalScale(domain, range, value) {
+  const step = (domain[1] - domain[0]) / range.length;
+  const idx = Math.floor((value - domain[0]) / step);
+  const clampIdx = Math.max(Math.min(idx, range.length - 1), 0);
+
+  return range[clampIdx];
+}
+
+function _linearScale(domain, range, value) {
+
+  return (value - domain[0]) / (domain[1] - domain[0]) * (range[1] - range[0]) + range[0];
+}
+
+export default class PointDensityGridLayer extends Layer {
+  initializeState() {
+    this.state = {
+      gridOffset: {latOffset: 0.0089, lonOffset: 0.0113},
+      layerData: [],
+      countRange: null,
+      pickedCell: null
+    };
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    if (changeFlags.dataChanged || _needsReProjectPoints(oldProps, props)) {
+      const {data, cellSize} = this.props;
+      const {gridOffset, layerData, countRange} = pointToDensityGridData(data, cellSize);
+
+      Object.assign(this.state, {gridOffset, layerData, countRange});
+    }
+  }
+
+  pick(opts) {
+    super.pick(opts);
+
+    const pickedCell = this.state.pickedCell;
+
+    Object.assign(opts.info, {
+      layer: this,
+      // override index with cell index
+      index: pickedCell ? pickedCell.index : -1,
+      picked: Boolean(pickedCell),
+      // override object with picked cell
+      object: pickedCell
+    });
+  }
+
+  _onHoverSublayer(info) {
+
+    this.state.pickedCell = info.picked && info.index > -1 ?
+      this.state.layerData[info.index] : null;
+  }
+
+  _onGetSublayerColor(cell) {
+    const {colorRange} = this.props;
+    const colorDomain = this.props.colorDomain || this.state.countRange;
+
+    return _ordinalScale(colorDomain, colorRange, cell.count);
+  }
+
+  _onGetSublayerElevation(cell) {
+    const {elevationRange} = this.props;
+    const elevationDomain = this.props.elevationDomain || [0, this.state.countRange[1]];
+    return _linearScale(elevationDomain, elevationRange, cell.count);
+  }
+
+  renderLayers() {
+    const {id} = this.props;
+
+    // Override user's onHover and onClick props
+    const handlers = {
```

Comment:
In my opinion defining this `handlers` variable does not make the code easier to read.

As opposed to just listing these in the `Object.assign` below with a comment in between.

We don't have multiple sublaysers like in `GeoJsonLayer`, so I'd consider merging the two objects.